charts/opentelemetry-demo/templates/_pod.tpl

```diff
@@ -41,25 +41,33 @@ Get Services Port Mapping
 {{- end }}
 
 {{/*
-Get Pod Env 
+Get Pod Env
 */}}
 {{- define "otel-demo.pod.env" -}}
 {{- $prefix := include "otel-demo.name" $ }}
 
+{{- if .default.enabled  }}
+{{- toYaml .default.env }}
+{{- end }}
+
 {{- if .env }}
-{{- toYaml .env }}
+{{ toYaml .env }}
```

Comment:
a component's individual environment variables should probably be merged with the defaults.  That way there are no duplicates and a individual component could change a value without having to change it for all other components.

Reply:
finished,please take a look

Reply:
.default.env and .env  are not a map but slice containing env maps. I will try to use build-in funciton to implement the merge